### PR TITLE
Refactor block production to have limited execution time and block size

### PR DESCRIPTION
**Pull request by bytemaster**
This should contain block size to 1000 trx.

### CODE REVIEW

File: libraries/chain/chain_controller.cpp [hunk 1]
```diff
@@ -241,6 +241,8 @@ processed_transaction chain_controller::_push_transaction(const signed_transacti
    if (!_pending_tx_session.valid())
       _pending_tx_session = _db.start_undo_session(true);
 
+   FC_ASSERT( _pending_transactions.size() < 1000, "too many pending transactions, try again later" );
```
Comment (brianjohnson5972) [inline libraries/chain/chain_controller.cpp]: If we make 1000 a config parameter in chain_controller, then we can adjust these during testing and not have to wait for a tag and recompile for the test net

Comment (heifner) [reply libraries/chain/chain_controller.cpp]: @brianjohnson5972 go ahead and prepare a PR with making these two configurable.

File: libraries/chain/chain_controller.cpp [hunk 2]
```diff
@@ -429,16 +366,14 @@ signed_block chain_controller::_generate_block(
       pending_block.producer_changes = get_global_properties().active_producers - new_schedule;
    }
 
+   auto end = fc::time_point::now();
+   ilog( "generation took ${x} ms", ("x", (end-start).count()/1000) );
```
Comment (heifner) [inline libraries/chain/chain_controller.cpp]: This is generating too many output statements causing the travis build to fail.

**Comment by heifner** [comment]
@brianjohnson5972 Since I added to this PR, please review at least my portion. I modified the timings on your rate limiting tests.

**Pull request merged**